fibonacci.py

```diff
@@ -0,0 +1,12 @@
+def fibonacci(n):
+    number_fibo = (n+1) * [None]
+
+    def recursive_fibonacci(x):
+        if number_fibo[x] is None:
+            number_fibo[x] = x if x <= 1 else recursive_fibonacci(x-1) + recursive_fibonacci(x-2)
+        return number_fibo[x]

```

Comment:
caching the partial results is not a bad idea, but it would cause memory issues really fast, if speed is that much of a concern, than the Binet formula might be preferable